### PR TITLE
Remove Evidence while keeping ABCI compatibility. 

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
Removes Evidence from Block struct. Comments out relevant code lines with:
```// Note: Temporarily remove Evidence #896```

for adding back in the future when needed.

Closes: #896

### CODE REVIEW

File: types/serialization.go [hunk 1]
```diff
@@ -183,7 +181,8 @@ func (d *Data) ToProto() *pb.Data {
 	return &pb.Data{
 		Txs:                    txsToByteSlices(d.Txs),
 		IntermediateStateRoots: d.IntermediateStateRoots.RawRootsList,
-		Evidence:               evidenceToProto(d.Evidence),
+		// Note: Temporarily remove Evidence #896
+		// Evidence:               evidenceToProto(d.Evidence),
```
Comment (tzdybal) [inline types/serialization.go:185]: This should be tested using actual cosmod-sdk app to ensure that cosmos-sdk is not panicing or failing validation somewhere.

**Review by tzdybal — COMMENTED** [review]
LGTM, but this code needs to be tested against cosmos-sdk to ensure compatibility.

**Review by tzdybal — DISMISSED** [review]

**Review by gupadhyaya — APPROVED** [review]

File: types/serialization.go (hunk 1 above)
Comment (gupadhyaya) [reply types/serialization.go:185]: tested gm tutorial with a sequencer and a full node in local-devnet and it does not panic. so we are good to go.

**Review by tzdybal — APPROVED** [review]

**Comment by nashqueue** [comment]
> LGTM, but this code needs to be tested against cosmos-sdk to ensure compatibility.

Was this tested @Manav-Aggarwal

**Comment by Manav-Aggarwal** [comment]
> > LGTM, but this code needs to be tested against cosmos-sdk to ensure compatibility.
> 
> Was this tested @Manav-Aggarwal

yes, see https://github.com/rollkit/rollkit/pull/915#discussion_r1191152985

**Pull request merged**